tests/test_endpoints.py

```diff
@@ -163,13 +163,13 @@ def test_startup(self):
         Endpoint._startup()
         assert Endpoint.from_name('test-endpoint') is not None
         assert Endpoint.from_name('test-endpoint').endpoint_name == 'test-endpoint'
-        assert Endpoint.from_name('test-endpoint').joined
```

Comment:
I wouldn't remove the .joined checks, since the method needs to remain for backwards compatibility. Just add the new checks with is_joined.